group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -1921,14 +1921,14 @@ private CompletableFuture<DeleteShareGroupOffsetsResponseData> handleDeleteShare
                 errorTopicResponses.add(
                     new DeleteShareGroupOffsetsResponseData.DeleteShareGroupOffsetsResponseTopic()
                         .setTopicId(topicData.topicId())
-                        .setTopicName(metadataImage.topics().topicIdToNameView().get(topicData.topicId()))
+                        .setTopicName(metadataImage.topicName(topicData.topicId()).orElse(null))
```

Comment:
Should you ever get null here? I do see that it matches the existing behavior.

Reply:
It probably wouldn't ever happen but returning null at least matches the current behavior and avoids throwing an exception if the Optional is empty